openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/Teams.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright 2022 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { Button, Col, Row, Space, Switch } from 'antd';
+import React, { FC } from 'react';
+import { Team, TeamType } from '../../generated/entity/teams/team';
+import TeamHierarchy from './TeamHierarchy';
+import './teams.less';
+
+interface TeamsProps {
+  showDeletedTeam: boolean;
+  onShowDeletedTeamChange: (checked: boolean) => void;
+  data: Team[];
+  onAddTeamClick: (value: boolean) => void;
+}
+
+const Teams: FC<TeamsProps> = ({
+  data,
+  showDeletedTeam,
+  onShowDeletedTeamChange,
+  onAddTeamClick,
+}) => {
+  const generateTeamsData = (teams: Team[]) => {
+    const orgnization = teams
+      .filter((team) => team.teamType === TeamType.Organization)
+      .map((team) => {
+        const children = team.children?.map((c) => {
+          const updatedChildren: Team =
+            teams.find((dd: Team) => dd.id === c.id) || ({} as Team);
+          if (updatedChildren.children?.length === 0) {
+            delete updatedChildren.children;
```

Comment:
Why are we deleting the children here?

Reply:
this will be going to replace once backend API comes!